packages/popup/src/pages/ImportPrivateKey/index.js

```diff
@@ -0,0 +1,137 @@
+import {useState, useEffect} from 'react'
+import {useHistory} from 'react-router-dom'
+import {useTranslation} from 'react-i18next'
+import {TextNav} from '../../components/index'
+import Button from '@cfxjs/component-button'
+import Input from '@cfxjs/component-input'
+import {CompWithLabel} from '../../components'
+import {useRPC} from '@cfxjs/use-rpc'
+import {request} from '../../utils'
+import {GET_ALL_ACCOUNT_GROUP, GET_PK_ACCOUNT_GROUP} from '../../constants'
+import useGlobalStore from '../../stores'
+import {useCreatedPasswordGuard} from '../../hooks'
+
+import {useSWRConfig} from 'swr'
+
+function ImportAccount() {
+  const {t} = useTranslation()
+  const history = useHistory()
+  const {mutate} = useSWRConfig()
+
+  const [name, setName] = useState('')
+  const [keygen, setKeygen] = useState('')
+  const [keygenErrorMessage, setKeygenErrorMessage] = useState('')
+  const [keygenNamePlaceholder, setKeygenNamePlaceholder] = useState('')
+  const [creatingAccount, setCreatingAccount] = useState(false)
+  const createdPassword = useGlobalStore(state => state.createdPassword)
+
+  const {data: keygenGroup} = useRPC(
+    [...GET_PK_ACCOUNT_GROUP],
+    {
+      type: 'pk',
+    },
+    {
+      fallbackData: [],
+    },
+  )
+
+  useCreatedPasswordGuard()
+  useEffect(() => {
+    setKeygenNamePlaceholder(`Account-${keygenGroup.length + 1}`)
+  }, [keygenGroup])
+
+  const walletValidatePrivateKey = keygen => {
+    setKeygenErrorMessage(keygen === '' ? 'Required!' : '')
+  }
+  const changeName = e => {
+    setName(e.target.value)
+  }
+  const changeKeygen = e => {
+    setKeygen(e.target.value)
+    walletValidatePrivateKey(e.target.value)
+  }
+  const dispatchMutate = () => {
+    mutate([...GET_ALL_ACCOUNT_GROUP])
+    mutate([...GET_PK_ACCOUNT_GROUP])
+  }
+  const importAccount = async () => {
+    if (
+      !creatingAccount &&
+      name.length <= 20 &&
+      !keygenErrorMessage &&
+      keygen
+    ) {
+      setCreatingAccount(true)
+      request('wallet_importPrivateKey', {
+        password: createdPassword,
+        nickname: name || keygenNamePlaceholder,
+        privateKey: keygen,
+      }).then(({error, res}) => {
+        setCreatingAccount(false)
+        if (res?.result) {
+          dispatchMutate()
+          history.push('/')
+        }
+        if (error?.message) {
+          setKeygenErrorMessage(error.message.split('\n')[0])
```

Comment:
should go to the error page, since it's allowd to import duplicate pk now

Reply:
?

Reply:
We can hang on to the question for now. Until the product manager gives a definitive answer